samples/java-spring-eventsourced-customer-registry/src/main/java/customer/domain/CustomerEvent.java

```diff
@@ -6,13 +6,13 @@
 
 public sealed interface CustomerEvent {
 
-  @TypeName("customer-created") // <1>
+  @TypeName("internal-customer-created") // <1>
```

Comment:
Why the prefix `internal` ?

Reply:
There is another sealed interface used for the public api.. and they had the same names. That causes an exception as these need to be unique. So I changed the internal ones to have that prefix.

Reply:
I'm curious to understand why they need to be unique. 

Are they handle in different code paths? Why the name clash?

Reply:
Oh yes, that's possible. Doesn't matter which components and how to get there. We always pass by the cache. 

Which makes me think if we should not have a namespace for the cache. A cache per component. For the user, there is not reason why they could not reuse a name. And can even be confusing when seeing such an error message. Worst, that's a runtime error.

Reply:
here it is https://github.com/lightbend/kalix-jvm-sdk/issues/1556